src/matlab2tikz.m

```diff
@@ -892,26 +892,52 @@ case guitypes()
 end
 % ==============================================================================
 function m2t = drawGridOfAxes(m2t, handle)
-    % draws the grids of an axes
-    %TODO: has{XYZ}Grid is always false without a good reason
-    hasXGrid = false;
-    hasYGrid = false;
-    hasZGrid = false;
-    if hasXGrid || hasYGrid || hasZGrid
-        matlabGridLineStyle = get(handle, 'GridLineStyle');
-        % Take over the grid line style in any case when in strict mode.
-        % If not, don't add anything in case of default line grid line style
-        % and effectively take Pgfplots' default.
-        defaultMatlabGridLineStyle = ':';
-        if m2t.cmdOpts.Results.strict ...
-                || ~strcmpi(matlabGridLineStyle,defaultMatlabGridLineStyle)
-            gls = translateLineStyle(matlabGridLineStyle);
+    % Draws the grids of an axis
+    % Check for major grids
+    hasGrid =  isOn(get(handle, 'XGrid')) ...
+            || isOn(get(handle, 'YGrid')) ...
+            || isOn(get(handle, 'ZGrid'));
+
+    % Check for minor grids
+    hasMinorGrid =  isOn(get(handle, 'XMinorGrid')) ...
+                 || isOn(get(handle, 'YMinorGrid')) ...
+                 || isOn(get(handle, 'ZMinorGrid'));
+
+    % Default matlab grid style ommited for brevity
+    % defaultGridLineStyle = ':';
```

Comment:
You should be able to get the same result with:

```matlab
[gridLineStyle, isDefault] = getAndCheckDefault('axes', handle, 'GridLineStyle', ':');
```

which should work for both HG1 and HG2. 

What that function does, is it first checks the MATLAB factory defaults for the relevant setting. If that setting is not found, it uses the `':'` as fallback. It retrieves the property from your object and `isDefault` will be true when the default value and the actual one are equal.

As far as I'm concerned, you can just pass in `':'` as a literal to the function, without giving it a name, just make sure you remove the commented-out code: it can only become outdated and then we have to dig around to see what to do with it.